lib/shape.ml

```diff
@@ -4,74 +4,82 @@ type point = { x : int; y : int }
 type line = { a : point; b : point }
 type rectangle = { c : point; length : int; width : int }
 type circle = { c : point; radius : int }
-type ellipse = {c : point; rx : int; ry: int}
+type ellipse = { c : point; rx : int; ry : int }
 type shape = Circle of circle | Rectangle of rectangle | Ellipse of ellipse | Line of line
 type shapes = shape list
 
-let dimensions = ref {x = 500; y = 500}
-let set_dimensions x y = 
-  dimensions := {x;y} 
+let dimensions = ref { x = 500; y = 500 }
+let set_dimensions x y =
+  dimensions := { x; y }
 
-let canvas_mid = { x = (!dimensions.x / 2); y = (!dimensions.y / 2)}
+let canvas_center = { x = 0; y = 0 }
```

Comment:
@nangahamandine i don't think you understood what I meant in the last review.

We need the old `canvas_mid` as it was.

All we need to do in this PR is to set the initialisation defaults to a center with (0, 0), and then when rendering use `denormalize` to take it into account.